src/Shared/TypeExtensions.cs

```diff
@@ -13,34 +13,82 @@ internal static partial class TypeExtensions
                                                                  BindingFlags.IgnoreCase |
                                                                  BindingFlags.DeclaredOnly;
 
+
```

Comment:
Thanks for adding comments.  Code comments are optional for internal/private class and/or method.